### DISCUSSION TITLE
Get() using WithTimeout() ctx

### DISCUSSION THREAD

**Discussion by aathan** [olric-data/olric#179] [Q&A]
What is the return value from DMap.Get() in the event a WithTimeout() ctx is used with the call? Does it return a non-nil error in that case? What is the recommended way to use that call, or olris in general, in a pseudo hard-realtime scenario where the calls like Get() should not hang forever?

**Comment by buraksezer** [comment]
Olric `v0.5.x` has context as a parameter, previous versions don't have any protection against being stuck. The latest version passes that context to the underlying Redis client. Here is the `Get` implementation of ClusterClient of v0.5.0.

```go
func (dm *ClusterDMap) Get(ctx context.Context, key string) (*GetResponse, error) {
	cmd := protocol.NewGet(dm.name, key).SetRaw().Command(ctx)
	rc, err := dm.clusterClient.smartPick(dm.name, key)
	if err != nil {
		return nil, err
	}
	err = rc.Process(ctx, cmd)
	if err != nil {
		return nil, processProtocolError(err)
	}
	return dm.makeGetResponse(cmd)
}
```

> Does it return a non-nil error in that case? 

It should return `context.DeadlineExceeded` in the case of context timeout.
